### ISSUE TITLE
Issue -1 - Email Verification Fix

### ISSUE THREAD

**Issue by Ramya-Ravisankar** [Ramya-Ravisankar/user_management#9]
Issue -1 - Email Verification Fix
Steps to reproduce:
1)	Access the User Management Swagger UI - http://localhost/docs
2)	Go to the Login and Registration section.
3)	Under the post register section, click on the try it out link.
4)	Under the request body, give a new user detail and click on execute.
5)	Now login into mailtrap.io and click on the verify email link for the above created new user.
6)	Error message displayed:
{"detail":[{"type":"uuid_parsing","loc":["path","user_id"],"msg":"Input should be a valid UUID, invalid character: expected an optional prefix of `urn:uuid:` followed by [0-9a-fA-F-], found `N` at 1","input":"None","ctx":{"error":"invalid character: expected an optional prefix of `urn:uuid:` followed by [0-9a-fA-F-], found `N` at 1"},"url":"https://errors.pydantic.dev/2.6/v/uuid_parsing"}]}

To fix the above issue: Modified the user_service.py and test_user_service.py

Solution from the above fix - The email verification will be sent only after the new user is committed to the DB. Also, the user role will be changed to authenticated only when the user role is anonymous, so that it does not affect other roles.

Code Changes: 
user_service.py and test_user_service.py -  Code Changes can be viewed in the below link

THE CHANGES RELATED TO THIS ISSUE HAS BEEN MERGED INTO MAIN.

https://github.com/Ramya-Ravisankar/user_management/pull/2/files
